choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerController.java

```diff
@@ -275,4 +513,121 @@ private void checkPlanRequest(final ChoreographerPlanRequestDTO request, final S
             throw new BadPayloadException("Plan name is null or blank.", HttpStatus.SC_BAD_REQUEST, origin);
```

Comment:
additional checks

- request null check
- firstActionName empty check
- action list not null or empty check
- action list contains an action with name specified in firstActionName
- and formal checks on actions and steps

- every name should be transformed (toLowercase().trim())